modules/text-area/react/spec/TextArea.spec.tsx

```diff
@@ -1,44 +1,81 @@
+import {render, fireEvent} from '@testing-library/react';
 import TextArea from '../lib/TextArea';
+import ReactDOMServer from 'react-dom/server';
 import {axe} from 'jest-axe';
 import FormField from '@workday/canvas-kit-react-form-field';
 
-describe('TextArea', () => {
-  const cb = jest.fn();
-  afterEach(() => {
-    cb.mockReset();
-  });
-  test('TextArea should spread extra props', () => {
-    const component = mount(<TextArea data-propspread="test" onChange={cb} />);
-    const container = component.at(0).getDOMNode();
-    expect(container.getAttribute('data-propspread')).toBe('test');
-    component.unmount();
-  });
-});
+const id = 'Test Text Area';
+const label = 'Test Text Area';
+const placeholder = 'Test Text Area';
+const value = 'Test Text Area';
 
-describe('Text Area Accessibility', () => {
+describe('Text Area', () => {
   const cb = jest.fn();
   afterEach(() => {
     cb.mockReset();
   });
-  test('Text Area in a FormField should pass axe DOM accessibility guidelines', async () => {
-    const html = ReactDOMServer.renderToString(
-      <FormField label="My Field" inputId="my-input-field">
-        <TextArea placeholder="Placeholder" value={'Hello'} onChange={jest.fn()} />;
-      </FormField>
-    );
-    expect(await axe(html)).toHaveNoViolations();
+
+  describe('when rendered', () => {
+    describe('with an placeholder', () => {
+      test('should render a text area with placeholder', () => {
+        const {getByPlaceholderText} = render(<TextArea onChange={cb} placeholder={placeholder} />);
+        expect(getByPlaceholderText(placeholder)).toHaveAttribute('placeholder', placeholder);
+      });
+    });
+
+    describe('with a id', () => {
+      test('should render a text area with a id', () => {
+        const {getByPlaceholderText} = render(
+          <TextArea id={id} onChange={cb} value={value} placeholder={placeholder} />
+        );
+        expect(getByPlaceholderText(placeholder)).toHaveAttribute('id', id);
+      });
+    });
+
+    describe('with a value', () => {
+      test('should render a text area with a value', () => {
+        const {getByDisplayValue} = render(<TextArea onChange={cb} value={value} />);
+        expect(getByDisplayValue(value)).toBeDefined();
+      });
+    });
+
+    describe('with disabled attribute', () => {
+      test('should render a disabled text area', () => {
+        const {getByDisplayValue} = render(
+          <TextArea onChange={cb} disabled={true} value={value} />
+        );
+        expect(getByDisplayValue(value)).toBeDisabled();
+      });
+    });
+
+    describe('wrapped in a FormField', () => {
+      test('should pass axe DOM accessibility guidelines', async () => {
+        const html = ReactDOMServer.renderToString(
+          <FormField label="My Field" inputId="my-text-area-field">
+            <TextArea id="my-text-area-field" onChange={cb} />
+          </FormField>
+        );
+        expect(await axe(html)).toHaveNoViolations();
```

Comment:
I think this type of test should be moved to Cypress, right? Accessibility runs audits on our stories and not our unit tests.

This request can be part of our broader refactor of testing where component specifications are written and not part of this PR

Reply:
This may be good to include in the Cypress section of the testing docs